tests/test_other.py

```diff
@@ -2545,6 +2545,27 @@ def test_fs_stream_proto(self):
       out = self.run_js('a.out.js', engine=engine)
       self.assertContained('File size: 724', out)
 
+  def test_node_emscripten_num_logical_cores(self):
+    # Test with node.js that the emscripten_num_logical_cores method is working
+    if NODE_JS not in JS_ENGINES:
+      self.skipTest("node engine required for this test")
```

Comment:
these lines can be removed - we must always have node.js (it is necessary for emcc.py anyhow).

Reply:
I removed these lines with commit https://github.com/emscripten-core/emscripten/pull/12008/commits/6741743f06bfda2c5aeff04a31dd7e5f9b4755bd, thanks!

(fwiw, there are still a few assumptions in this file that Node.js is _not_ always present, just grep for `if NODE_JS in JS_ENGINES:`  and `if NODE_JS not in JS_ENGINES:`)

Reply:
Yeah... I think `NODE_JS` has to always exist, but `JS_ENGINES` is the list of engines to use during *testing* and that might not contains `NODE_JS`.. for example if you want to run the tests under `D8` you would set `JS_ENGINES` to `[D8]`, even though NODE_JS must exists to do the compilation phases.